Metalama.Backstage/src/Metalama.Backstage/UserInterface/Rss/RssClient.cs

```diff
@@ -0,0 +1,260 @@
+        {
+            // Check preconditions.
+            if ( configuration.PreferredFeed == RssFeed.None )
+            {
+                this._logger.Trace?.Log( "The RSS client has been disabled." );
+
+                return;
+            }
+
+            if ( !this._telemetryConfigurationService.IsEnabled( TelemetryScenario.Rss ) )
+            {
+                this._logger.Trace?.Log( "Telemetry is disabled. Do not fetch news." );
+
+                return;
+            }
+
+            if ( configuration.LastFetchTime == null )
+            {
+                this._logger.Trace?.Log( "This is the first time RssClient fetches news. Do not report any past news." );
+
+                // Never display past items upon first fetch.
+                this._configurationManager.Update<RssClientConfiguration>(
+                    c => c with { LastFetchTime = this._dateTimeProvider.UtcNow, PreferredFeed = RssFeed.Briefs } );
+
+                return;
+            }
+            else if ( configuration.LastFetchTime.Value.AddDays( 1 ) > this._dateTimeProvider.UtcNow )
+            {
+                this._logger.Trace?.Log( $"News were already checked on {configuration.LastFetchTime}." );
+
+                return;
+            }
+        }
+
+        // Select feed URL.
+        var url = configuration.PreferredFeed switch
+        {
+            null or RssFeed.Briefs => BriefsUrl,
+            RssFeed.Posts => PostsUrl,
+            _ => null
+        };
+
+        if ( url == null )
+        {
+            this._logger.Warning?.Log( $"Invalid preferred feed: {configuration.PreferredFeed}. Skipping." );
+
+            return;
+        }
+
+        try
+        {
+            // Fetch content.
+            var httpClient = this._httpClientFactory.Create();
+            var response = await httpClient.GetAsync( url );
+
+            if ( !response.IsSuccessStatusCode )
+            {
+                this._logger.Trace?.Log(
+                    $"Cannot get '{url}': {response.ReasonPhrase}." );
+
+                return;
+            }
+            
+            var content = await response.Content.ReadAsStringAsync();
+
+            // Try to parse the item.
+            if ( !this.TryParseContent( content, out var title, out var link, out var pubDate ) )
+            {
+                return;
+            }
+
+            // Only notify if the item was published after the last fetch time.
+            if ( configuration.LastFetchTime != null && pubDate != null && pubDate.Value <= configuration.LastFetchTime!.Value )
+            {
+                this._logger.Trace?.Log(
+                    $"Item published on {pubDate} is not newer than last fetch time {configuration.LastFetchTime}. Skipping notification." );
+
+                return;
+            }
+
+            // Update last fetch time.
+            this._configurationManager.Update<RssClientConfiguration>( c => c with { LastFetchTime = this._dateTimeProvider.UtcNow } );
```

Comment:
This should probably use the time of the last observed item instead of local time; otherwise, we may miss some items in an edge case. Not important though.